core/lib/src/request/request.rs

```diff
@@ -934,6 +1012,10 @@ impl<'r> Request<'r> {
             if self.accept().is_none() || replace {
                 self.state.accept = Storage::new();
             }
+        } else if self.rocket().config.proxy_proto_header.as_ref().map_or(false, |h| h == name) {
+            if !self.cookies().state.secure || replace {
+                self.cookies_mut().state.secure = self.context_is_likely_secure();
```

Comment:
I think this will not work, as the header is not added te the header map yet. Therefore, `context_is_likely_secure` will call `proxy_proto` which will always return `None`, am I right?

Reply:
This method is called any time a header is added or replaced in the header map. If it starts in the map, then `state.secure` == `true`. If it's added later, than this method will correct the value. That's the idea, at least.

Reply:
Fixed in 678b7c1f8dfe59d43a6295c0221d102ba720d520 (by you)